CMakeLists.txt

```diff
@@ -1063,8 +1063,11 @@ if(HAVE_FSETXATTR)
 endif()
 
 check_symbol_exists(sa_family_t    "${CURL_INCLUDES}" HAVE_SA_FAMILY_T)
+check_symbol_exists(ADDRESS_FAMILY "${CURL_INCLUDES}" HAVE_ADDRESS_FAMILY)
 if(HAVE_SA_FAMILY_T)
   add_definitions(-DCURL_SA_FAMILY_T=sa_family_t)
+elseif(HAVE_ADDRESS_FAMILY)
+  add_definitions(-DCURL_SA_FAMILY_T=ADDRESS_FAMILY)
```

Comment:
Still, you should not set defines with `-D`. You should set them like we set all other defines for cmake: with `set()` and add it as a template to `lib/curl_config.h.cmake`

Compare how other defines are set. For example `HAVE_PTHREAD_H` or `HAVE_SIGSETJMP`